clients/kafka-common/src/main/java/org/eclipse/hono/kafka/client/tracing/KafkaTracingHelper.java

```diff
@@ -0,0 +1,134 @@
+ *
+ */
+// TODO align with Kafka tracing support in Vert.x 4.0
+public final class KafkaTracingHelper {
+
+    /**
+     * An OpenTracing tag that contains the offset of a Kafka record.
+     */
+    public static final LongTag TAG_OFFSET = new LongTag("offset");
+
+    /**
+     * An OpenTracing tag that contains the partition of a Kafka record.
+     */
+    public static final IntTag TAG_PARTITION = new IntTag("partition");
+
+    /**
+     * An OpenTracing tag that contains the timestamp of a Kafka record.
+     */
+    public static final LongTag TAG_TIMESTAMP = new LongTag("timestamp");
+
+    private KafkaTracingHelper() {
+        // prevent instantiation
+    }
+
+    /**
+     * Creates a new <em>OpenTracing</em> span to trace producing messages to Kafka.
+     * <p>
+     * The returned span will already contain the following tags:
+     * <ul>
+     * <li>{@link Tags#COMPONENT} - set to <em>hono-client-kafka</em></li>
+     * <li>{@link Tags#MESSAGE_BUS_DESTINATION} - set to {@code To_<topic>}</li>
+     * <li>{@link Tags#SPAN_KIND} - set to {@link Tags#SPAN_KIND_PRODUCER}</li>
+     * <li>{@link Tags#PEER_SERVICE} - set to <em>kafka</em></li>
+     * </ul>
+     *
+     * @param tracer The Tracer to use.
+     * @param topic The topic from which the operation name is derived.
+     * @param referenceType The type of reference towards the span context.
+     * @param parent The span context to set as parent and to derive the sampling priority from (may be null).
+     * @return The new span.
+     * @throws NullPointerException if tracer or topic is {@code null}.
+     */
+    public static Span newProducerSpan(final Tracer tracer, final HonoTopic topic, final String referenceType,
+            final SpanContext parent) {
+        Objects.requireNonNull(tracer);
+        Objects.requireNonNull(topic);
+        Objects.requireNonNull(referenceType);
+
+        return TracingHelper.buildSpan(tracer, parent, "To_" + topic.toString(), referenceType)
+                .ignoreActiveSpan()
+                .withTag(Tags.COMPONENT.getKey(), "hono-client-kafka")
+                .withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_PRODUCER)
+                .withTag(Tags.MESSAGE_BUS_DESTINATION.getKey(), topic.toString())
+                .withTag(Tags.PEER_SERVICE.getKey(), "kafka")
+                .start();
+    }
+
+    /**
+     * Sets tags from record metadata.
+     * <p>
+     * It sets the following tags:
+     * <ul>
+     * <li>{@link #TAG_OFFSET}</li>
+     * <li>{@link #TAG_PARTITION}</li>
+     * <li>{@link #TAG_TIMESTAMP}</li>
+     * </ul>
+     * <p>
+     * <em>It does not set the topic, as this is expected to be already already set.</em>
+     *
+     * @param span The span to set the tags on.
+     * @param recordMetadata The record metadata.
+     */
+    public static void setRecordMetadataTags(final Span span, final RecordMetadata recordMetadata) {
```

Comment:
IMHO these values should be logged to a span but should not be set as tags ... I doubt that you will filter/search spans based on these values, will you?

Reply:
I'm not sure. I aimed for compatibility with https://github.com/opentracing-contrib/java-kafka-client which does set _partition_ and _offset_ as tags. 
I just learned that Vert.x 4 will soon be released with support for OpenTracing. We may want Hono to take advantage of those APIs. So maybe this implementation should be aligned with the tracing that Vert.x's Kafka client will produce, which seems to be this https://github.com/vert-x3/vertx-kafka-client/blob/master/src/main/java/io/vertx/kafka/client/common/tracing/TraceTags.java
WDYT?